Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -157,21 +153,69 @@ public QuestionRepository(TrappistDbContext dbContext)
 
         public async Task UpdateSingleMultipleAnswerQuestionAsync(int questionId, QuestionAC questionAC, string userId)
         {
-            var updatedQuestion = Mapper.Map<QuestionDetailAC, Question>(questionAC.Question);
-            var options = questionAC.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption;
+            var updatedQuestion = await _dbContext.Question.FindAsync(questionId);
+            await _dbContext.Entry(updatedQuestion).Reference(x => x.SingleMultipleAnswerQuestion).LoadAsync();
+            await _dbContext.Entry(updatedQuestion.SingleMultipleAnswerQuestion).Collection(x => x.SingleMultipleAnswerQuestionOption).LoadAsync();
+            var option = updatedQuestion.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption;
+            var updatedOption = questionAC.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption;
 
-            //Update common Question details
-            updatedQuestion.Id = questionId;
+            Mapper.Map(questionAC.Question, updatedQuestion);
+            Mapper.Map(questionAC.SingleMultipleAnswerQuestion, updatedQuestion.SingleMultipleAnswerQuestion);
+            updatedQuestion.SingleMultipleAnswerQuestion.UpdateDateTime = DateTime.UtcNow;
             updatedQuestion.UpdatedByUserId = userId;
-            _dbContext.Question.Update(updatedQuestion);
             await _dbContext.SaveChangesAsync();
 
-            //Update single/multiple Question and option
-            _dbContext.SingleMultipleAnswerQuestionOption.RemoveRange(await _dbContext.SingleMultipleAnswerQuestionOption.Where(x => x.SingleMultipleAnswerQuestionID == questionId).ToListAsync());
-            await _dbContext.SaveChangesAsync();
-            options.ForEach(x => x.SingleMultipleAnswerQuestionID = questionId);
-            _dbContext.SingleMultipleAnswerQuestionOption.AddRange(options);
-            await _dbContext.SaveChangesAsync();
+            using (var transaction = _dbContext.Database.BeginTransaction())
+            {
+                var optionToUpdate = updatedOption.Where(x => option.Any(y => y.Id == x.Id)).ToList();
+                var optionToDelete = option.Where(x => !updatedOption.Any(y => y.Id == x.Id)).ToList();
+                var optionToAdd = updatedOption.Where(x => !option.Any(y => y.Id == x.Id)).ToList();
+
+                //Remove options from updated question
+                _dbContext.SingleMultipleAnswerQuestionOption.RemoveRange(optionToDelete);
+                await _dbContext.SaveChangesAsync();
+
+                //Add new options to updated question
+                optionToAdd.ForEach(x =>
+                {
+                    x.SingleMultipleAnswerQuestionID = updatedQuestion.SingleMultipleAnswerQuestion.Id;
+                    x.Id = 0;
```

Comment:
can you please check that by omitting this statement, does it creates any effect?